### ISSUE TITLE
[sqlancer] query result error

### ISSUE THREAD

**Issue by wanpengfei-git** [StarRocks/starrocks#6552]
### Steps to reproduce the behavior (Required)
```
DROP DATABASE IF EXISTS g0;
CREATE DATABASE g0;
USE g0;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATETIME NOT NULL ,c_0_1 DATE NOT NULL ,c_0_2 DECIMAL(12, 2)  NOT NULL ,c_0_3 VARCHAR(21) NOT NULL ,c_0_4 DATE NULL ,c_0_5 DATETIME NOT NULL ,c_0_6 SMALLINT NOT NULL ,c_0_7 DECIMAL(32, 16)  NULL ,c_0_8 DECIMAL(7, 5)  NULL ,c_0_9 DATE NOT NULL ,c_0_10 SMALLINT SUM NOT NULL ,c_0_11 VARCHAR(1) MAX NOT NULL ,c_0_12 PERCENTILE PERCENTILE_UNION NOT NULL ,c_0_13 INT REPLACE NULL ,c_0_14 HLL HLL_UNION NOT NULL ,c_0_15 HLL HLL_UNION NULL ) AGGREGATE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6,c_0_7,c_0_8,c_0_9) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 365 day)) DISTRIBUTED BY HASH (c_0_4,c_0_3,c_0_8,c_0_6,c_0_1) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 DECIMAL(21, 12)  NULL ,c_1_1 DATE NOT NULL ,c_1_2 DATE NULL ,c_1_3 CHAR(11) NOT NULL ,c_1_4 SMALLINT NOT NULL ,c_1_5 ARRAY<STRING> NULL , INDEX index1_c_1_1(`c_1_1`) USING BITMAP, INDEX index1_c_1_2(`c_1_2`) USING BITMAP) DUPLICATE KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4) PARTITION BY RANGE(c_1_2) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 136 day)) DISTRIBUTED BY HASH (c_1_2,c_1_1,c_1_4,c_1_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t2 (c_2_0 DATE NOT NULL ,c_2_1 BOOLEAN NOT NULL ,c_2_2 BOOLEAN NOT NULL ,c_2_3 CHAR(1) NULL ,c_2_4 DECIMAL(12, 11)  NOT NULL ,c_2_5 TINYINT NOT NULL ,c_2_6 CHAR(11) NULL ,c_2_7 LARGEINT NOT NULL ,c_2_8 BOOLEAN NOT NULL ,c_2_9 FLOAT NULL ,c_2_10 BOOLEAN NULL ,c_2_11 DOUBLE NOT NULL ) UNIQUE KEY (c_2_0,c_2_1,c_2_2,c_2_3,c_2_4) PARTITION BY RANGE(c_2_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 547 day)) DISTRIBUTED BY HASH (c_2_3) BUCKETS 3 properties("replication_num"="3", "bloom_filter_columns"="c_2_3") ;
INSERT INTO t2 VALUES ('2010-01-20', false, false, '', 3E-11, 5, 'goLIv', 667989270, false, 0.5568719687185081, false, 0.0036183718442874824), ('2010-01-03', true, true, '', 6E-11, 1, 'L', -1851869962, true, 0.7646852852350777, true, 0.4841586377904291), ('2010-01-24', true, true, '', 5E-11, 1, NULL, -1589966817, false, 0.7853603434893112, false, 0.372934994739327), ('2010-01-02', true, false, '', 1E-11, 3, '3fx0t', -1589966817, true, 0.7330598185691354, true, 0.32988838624937744);
INSERT INTO t1 VALUES (NULL, '2010-01-05', NULL, 'i', 4, ["v8", "", "", "fTF", "96f", "", "3Sj", "", "wFlxCQ", "G", "Oe", "XAt", "M0bNKdJY"]);
INSERT INTO t1 VALUES (0.000609795184, '2010-01-13', '2010-01-25', 'I26QTYt', 0, ["pHOH", "thWX", "G4kmJ", "", "XZXM3", "", "", "G0", "XhaB6x7h", "IN"]), (0.000945901965, '2010-01-17', '2010-01-03', 'C', 3, ["", "T9MNd", "3WFF9B", "Z4IWwSi3A", "7Z0MUe8", "Jlufml2I", "zkL", "Edjo", "", "ITQz", "q3RVYTAzK", "", "", "XC"]);
INSERT INTO t1 VALUES (0.000654449504, '2010-01-18', '2010-01-24', '', 0, ["", "i1E8uKU", "", "jE", "", "", "vk9xsn"]), (0.000888388596, '2010-01-03', '2010-01-15', '', 7, ["61Ry", "4", "qVZe6", "2Gj", ""]);
INSERT INTO t1 VALUES (0.000923497578, '2010-01-25', '2010-01-18', 'oa6', 2, ["b", "S", "", "", "bp39CYLx", "dfq7ik", "", "w", ""]), (0.000192438670, '2010-01-14', '2010-01-20', 'Eq', 15, ["kylAaHhj", "kWnSl5", "", "K", "faD", "ylO9I"]);
INSERT INTO t2 VALUES ('2010-01-07', false, true, '', 6E-11, 1, '', -1510872359, false, 0.32988838624937744, false, 0.5568719687185081), ('2010-01-02', false, false, '', 3E-11, 2, 'Jqo4x1', 368055115, false, 0.17539408127999512, false, 0.31938662350969993);
INSERT INTO t1 VALUES (0.000046922934, '2010-01-02', '2010-01-23', 'M', 4, ["U", "oR", ""]);
INSERT INTO t0 VALUES ('2010-01-18 08:00:14', '2010-01-29', 88708561.18, 'pVBgZ', '2010-01-31', '2010-01-25 08:00:14', 0, 0.9920157330455474, 0.00072, '2010-01-13', 2, '', percentile_hash('-244182382'), NULL, hll_hash('il2yp'), hll_hash('1Pg4')), ('2010-01-11 08:00:14', '2010-01-07', 34776259.37, 'Sk', '2010-01-22', '2010-01-08 08:00:14', 14, 0.8359755521686707, 0.00009, '2010-01-25', 8, '', percentile_hash('-1510872359'), 18, hll_hash('0NzQ'), hll_hash('7r9')), ('2010-01-27 08:00:14', '2010-01-14', 89451599.06, '', '2010-01-30', '2010-01-05 08:00:14', 15, 0.2857772526200164, 0.00035, '2010-01-10', 1, '', percentile_hash('231646912'), 21, hll_hash('MkYC'), hll_hash('bknhh8'));
INSERT INTO t1 VALUES (0.000520864713, '2010-01-28', '2010-01-24', '', 4, ["1MSFGCr4B", "", "0", "TDw", "", "", "HIkK0Kdr", "id", "Vk"]);
INSERT INTO t0 VALUES ('2009-12-31 08:00:14', '2010-01-14', 44922818.51, 'q', '2010-01-11', '2010-01-14 08:00:14', 13, 0.6323239221009253, 0.00049, '2010-01-26', 4, '', percentile_hash('-1589966817'), 30, hll_hash(''), hll_hash('r'));
INSERT INTO t2 VALUES ('2010-01-25', true, true, '', 6E-11, 3, '', -2811186, false, 0.022134471116799337, false, 0.4678934830977648), ('2009-12-31', false, false, '', 6E-11, 6, '', -749554386, false, NULL, false, 0.17850587952258046);
INSERT INTO t0 VALUES ('2010-01-11 08:00:14', '2010-01-23', 93461682.53, '', '2010-01-13', '2010-01-02 08:00:14', 12, 0.3870702295009981, 0.00084, '2010-01-01', 13, '', percentile_hash('1046284006'), 5, hll_hash('iX'), hll_hash('')), ('2010-01-31 08:00:14', '2010-01-06', 79715717.81, 'L', '2010-01-12', '2010-01-20 08:00:14', 3, 0.4769629938457853, 0.00049, '2010-01-27', 7, '', percentile_hash('-1547215862'), 17, hll_hash(''), hll_hash('v90'));
INSERT INTO t2 VALUES ('2010-01-30', true, false, '', 8E-11, 0, 'Y', -1262301962, true, 0.8574681567504242, true, -1.547215862E9);
INSERT INTO t1 VALUES (NULL, '2009-12-31', '2010-01-31', '', 12, ["t", "", "n7Z", "a", "cD4C", "", "CS", "", ""]), (0.000383065786, '2010-01-21', '2010-01-23', '', 0, ["n6", "", "q0ce"]), (0.000269245880, '2010-01-19', '2010-01-20', '', 14, ["PnAf", "p"]);
INSERT INTO t0 VALUES ('2010-01-23 08:00:14', '2010-01-22', 40833603.14, '', '2010-01-20', '2010-01-16 08:00:14', 4, 0.9221720951313054, 0.00011, '2010-01-15', 8, '', percentile_hash('1108168492'), NULL, hll_hash(''), hll_hash('9c'));
TRUNCATE TABLE t2;
INSERT INTO t0 VALUES ('2010-01-14 08:00:14', '2010-01-01', 90798662.07, 'r', '2010-01-25', '2010-01-03 08:00:14', 1, 0.4845524537530551, 0.00057, '2010-01-21', 10, '', percentile_hash('231646912'), 29, hll_hash('Q'), hll_hash('4LHb')), ('2010-01-01 08:00:14', '2010-01-04', 1259011.90, 'kNWjUtcQY', '2010-01-21', '2010-01-17 08:00:14', 10, 0.9480661155634287, 0.00067, '2010-01-14', 10, '', percentile_hash('-1967063481'), 3, hll_hash('5'), hll_hash('mk'));
INSERT INTO t2 VALUES ('2010-01-24', true, false, NULL, 4E-11, 5, 'pzN', 237676399, false, 0.8860536216025391, false, 0.8547913695003166);
INSERT INTO t2 VALUES ('2010-01-29', false, true, '', 8E-11, 4, '', -558772101, false, -2811186.0, true, -1.589966817E9), ('2010-01-06', false, false, '', 8E-11, 7, 'Zz', 937425836, false, NULL, true, 0.5787180213564334);
INSERT INTO t1 VALUES (0.000846696052, '2010-01-17', '2010-01-21', '', 6, ["feIzfd", "cHlCSa0", "Z9", "3zR8ljeJ", "Ni", "xiI", "", "ML0Fb8rh", "U", "s4", "hDA", "koz", "R"]), (0.000609622422, '2010-01-31', '2010-01-17', 'j0', 10, ["", "5bB", "h2Fi", "e59XO", "", "OFzBSll7N", "cLbb", "C6", "gxI", "0G", "xxuL7", "", ""]);
INSERT INTO t2 VALUES ('2010-01-28', true, true, '', 2E-11, 6, '8X', -1510872359, false, NULL, false, 0.3504308799997373), ('2010-01-04', false, true, '', 8E-11, 1, 'zwc', -1589966817, true, 0.04958033242893145, true, -1.262301962E9);
-- SELECT subt0.c_0_10, subt0.c_0_9, subt0.c_0_2, t1_7.c_1_3 FROM  t1 AS t1_7  LEFT JOIN (SELECT t0_6.c_0_0, t0_6.c_0_1, t0_6.c_0_2, t0_6.c_0_3, t0_6.c_0_4, t0_6.c_0_5, t0_6.c_0_6, t0_6.c_0_7, t0_6.c_0_8, t0_6.c_0_9, t0_6.c_0_10, t0_6.c_0_11, t0_6.c_0_13 FROM t0 AS t0_6 WHERE ((('')||((('')||(''))))) IN ('', '', '') ) subt0 ON t1_7.c_1_4 = subt0.c_0_10 AND -1366265728 > subt0.c_0_10 GROUP BY subt0.c_0_10, subt0.c_0_9, subt0.c_0_2, t1_7.c_1_3;
-- cardinality: 8
-- SELECT subt0.c_0_10, subt0.c_0_9, subt0.c_0_2, t1_7.c_1_3 FROM  t1 AS t1_7  LEFT JOIN (SELECT t0_6.c_0_0, t0_6.c_0_1, t0_6.c_0_2, t0_6.c_0_3, t0_6.c_0_4, t0_6.c_0_5, t0_6.c_0_6, t0_6.c_0_7, t0_6.c_0_8, t0_6.c_0_9, t0_6.c_0_10, t0_6.c_0_11, t0_6.c_0_13 FROM t0 AS t0_6 WHERE ((('')||((('')||(''))))) IN ('', '', '') ) subt0 ON t1_7.c_1_4 = subt0.c_0_10 AND -1366265728 > subt0.c_0_10 WHERE (subt0.c_0_8) IN (0.00003, 0.00039)  GROUP BY subt0.c_0_10, subt0.c_0_9, subt0.c_0_2, t1_7.c_1_3 UNION SELECT subt0.c_0_10, subt0.c_0_9, subt0.c_0_2, t1_7.c_1_3 FROM  t1 AS t1_7  LEFT JOIN (SELECT t0_6.c_0_0, t0_6.c_0_1, t0_6.c_0_2, t0_6.c_0_3, t0_6.c_0_4, t0_6.c_0_5, t0_6.c_0_6, t0_6.c_0_7, t0_6.c_0_8, t0_6.c_0_9, t0_6.c_0_10, t0_6.c_0_11, t0_6.c_0_13 FROM t0 AS t0_6 WHERE ((('')||((('')||(''))))) IN ('', '', '') ) subt0 ON t1_7.c_1_4 = subt0.c_0_10 AND -1366265728 > subt0.c_0_10 WHERE (NOT ((subt0.c_0_8) IN (0.00003, 0.00039) )) GROUP BY subt0.c_0_10, subt0.c_0_9, subt0.c_0_2, t1_7.c_1_3 UNION SELECT subt0.c_0_10, subt0.c_0_9, subt0.c_0_2, t1_7.c_1_3 FROM  t1 AS t1_7  LEFT JOIN (SELECT t0_6.c_0_0, t0_6.c_0_1, t0_6.c_0_2, t0_6.c_0_3, t0_6.c_0_4, t0_6.c_0_5, t0_6.c_0_6, t0_6.c_0_7, t0_6.c_0_8, t0_6.c_0_9, t0_6.c_0_10, t0_6.c_0_11, t0_6.c_0_13 FROM t0 AS t0_6 WHERE ((('')||((('')||(''))))) IN ('', '', '') ) subt0 ON t1_7.c_1_4 = subt0.c_0_10 AND -1366265728 > subt0.c_0_10 WHERE (((subt0.c_0_8) IN (0.00003, 0.00039) ) IS NULL) GROUP BY subt0.c_0_10, subt0.c_0_9, subt0.c_0_2, t1_7.c_1_3;
-- cardinality: 7

```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - mysql> select current_version();
+----------------------+
| current_version()    |
+----------------------+
| MAIN-RELEASE 351ca95 |
+----------------------+
- cluster: sqlancer_01_release

**Comment by liuyehcf** [comment]
second query can be simplified to 

```sql
SELECT subt0.c_0_10,
       subt0.c_0_9,
       subt0.c_0_2,
       t1_7.c_1_3
FROM t1 AS t1_7
LEFT JOIN
  (SELECT t0_6.c_0_2,
          t0_6.c_0_8,
          t0_6.c_0_9,
          t0_6.c_0_10
   FROM t0 AS t0_6
   WHERE '' IN ('') ) subt0 ON t1_7.c_1_4 = subt0.c_0_10
AND -1366265728 > subt0.c_0_10
WHERE (subt0.c_0_8) IN (0.00003,
                        0.00039)
GROUP BY subt0.c_0_10,
         subt0.c_0_9,
         subt0.c_0_2,
         t1_7.c_1_3
UNION
SELECT subt0.c_0_10,
       subt0.c_0_9,
       subt0.c_0_2,
       t1_7.c_1_3
FROM t1 AS t1_7
LEFT JOIN
    (SELECT t0_6.c_0_2,
          t0_6.c_0_8,
          t0_6.c_0_9,
          t0_6.c_0_10
   FROM t0 AS t0_6
   WHERE '' IN ('') ) subt0 ON t1_7.c_1_4 = subt0.c_0_10
AND -1366265728 > subt0.c_0_10
WHERE (NOT ((subt0.c_0_8) IN (0.00003,
                              0.00039)))
GROUP BY subt0.c_0_10,
         subt0.c_0_9,
         subt0.c_0_2,
         t1_7.c_1_3
UNION
SELECT subt0.c_0_10,
       subt0.c_0_9,
       subt0.c_0_2,
       t1_7.c_1_3
FROM t1 AS t1_7
LEFT JOIN
    (SELECT t0_6.c_0_2,
          t0_6.c_0_8,
          t0_6.c_0_9,
          t0_6.c_0_10
   FROM t0 AS t0_6
   WHERE '' IN ('') ) subt0 ON t1_7.c_1_4 = subt0.c_0_10
AND -1366265728 > subt0.c_0_10
WHERE (((subt0.c_0_8) IN (0.00003,
                          0.00039)) IS NULL)
GROUP BY subt0.c_0_10,
         subt0.c_0_9,
         subt0.c_0_2,
         t1_7.c_1_3;
```

**Comment by liuyehcf** [comment]
Fixed by #6506

**Comment by liuyehcf** [comment]
The right plan

```
+--------------------------------------------------------------------------+
| Explain String                                                           |
+--------------------------------------------------------------------------+
| PLAN FRAGMENT 0                                                          |
|  OUTPUT EXPRS:17: c_0_10 | 16: c_0_9 | 9: c_0_2 | 4: c_1_3 | 23: count   |
|   PARTITION: UNPARTITIONED                                               |
|                                                                          |
|   RESULT SINK                                                            |
|                                                                          |
|   9:EXCHANGE                                                             |
|                                                                          |
| PLAN FRAGMENT 1                                                          |
|  OUTPUT EXPRS:                                                           |
|   PARTITION: HASH_PARTITIONED: 17: c_0_10, 16: c_0_9, 9: c_0_2, 4: c_1_3 |
|                                                                          |
|   STREAM DATA SINK                                                       |
|     EXCHANGE ID: 09                                                      |
|     UNPARTITIONED                                                        |
|                                                                          |
|   8:AGGREGATE (merge finalize)                                           |
|   |  output: count(23: count)                                            |
|   |  group by: 17: c_0_10, 16: c_0_9, 9: c_0_2, 4: c_1_3                 |
|   |                                                                      |
|   7:EXCHANGE                                                             |
|                                                                          |
| PLAN FRAGMENT 2                                                          |
|  OUTPUT EXPRS:                                                           |
|   PARTITION: HASH_PARTITIONED: 17: c_0_10                                |
|                                                                          |
|   STREAM DATA SINK                                                       |
|     EXCHANGE ID: 07                                                      |
|     HASH_PARTITIONED: 17: c_0_10, 16: c_0_9, 9: c_0_2, 4: c_1_3          |
|                                                                          |
|   6:AGGREGATE (update serialize)                                         |
|   |  STREAMING                                                           |
|   |  output: count(*)                                                    |
|   |  group by: 17: c_0_10, 16: c_0_9, 9: c_0_2, 4: c_1_3                 |
|   |                                                                      |
|   5:Project                                                              |
|   |  <slot 4> : 4: c_1_3                                                 |
|   |  <slot 9> : 9: c_0_2                                                 |
|   |  <slot 16> : 16: c_0_9                                               |
|   |  <slot 17> : 17: c_0_10                                              |
|   |                                                                      |
|   4:HASH JOIN                                                            |
|   |  join op: RIGHT OUTER JOIN (PARTITIONED)                             |
|   |  hash predicates:                                                    |
|   |  colocate: false, reason:                                            |
|   |  equal join conjunct: 17: c_0_10 = 5: c_1_4                          |
|   |                                                                      |
|   |----3:EXCHANGE                                                        |
|   |                                                                      |
|   1:EXCHANGE                                                             |
|                                                                          |
| PLAN FRAGMENT 3                                                          |
|  OUTPUT EXPRS:                                                           |
|   PARTITION: RANDOM                                                      |
|                                                                          |
|   STREAM DATA SINK                                                       |
|     EXCHANGE ID: 03                                                      |
|     HASH_PARTITIONED: 5: c_1_4                                           |
|                                                                          |
|   2:OlapScanNode                                                         |
|      TABLE: t1                                                           |
|      PREAGGREGATION: ON                                                  |
|      partitions=1/33                                                     |
|      rollup: t1                                                          |
|      tabletRatio=3/3                                                     |
|      tabletList=12020,12022,12024                                        |
|      cardinality=13                                                      |
|      avgRowSize=2.0                                                      |
|      numNodes=0                                                          |
|                                                                          |
| PLAN FRAGMENT 4                                                          |
|  OUTPUT EXPRS:                                                           |
|   PARTITION: RANDOM                                                      |
|                                                                          |
|   STREAM DATA SINK                                                       |
|     EXCHANGE ID: 01                                                      |
|     HASH_PARTITIONED: 17: c_0_10                                         |
|                                                                          |
|   0:OlapScanNode                                                         |
|      TABLE: t0                                                           |
|      PREAGGREGATION: OFF. Reason: Predicates include the value column    |
|      PREDICATES: CAST(17: c_0_10 AS INT) < -1366265728                   |
|      partitions=1/13                                                     |
|      rollup: t0                                                          |
|      tabletRatio=3/3                                                     |
|      tabletList=11817,11819,11821                                        |
|      cardinality=4                                                       |
|      avgRowSize=13.0                                                     |
|      numNodes=0                                                          |
+--------------------------------------------------------------------------+
```


The wrong plan

```
```
